### PR TITLE
set redis role label

### PR DESCRIPTION
**Pull request by klnchu**
Hi,

I found no services file about master/slave in the sample folder,  I want to use POD Label to select Redis role to do Service(master/slave).

Changes proposed on the PR:
-  Create master/slave label in pod.

### CODE REVIEW

File: operator/redisfailover/service/heal.go [hunk 1]
```diff
@@ -79,6 +81,28 @@ func (r *RedisFailoverHealer) SetMasterOnAll(masterIP string, rf *redisfailoverv
 	return nil
 }
 
+// SetRoleLable sets role tag for statful pod.
+func (r *RedisFailoverHealer) SetRoleLable(masterIP string, rf *redisfailoverv1alpha2.RedisFailover) error {
```
Comment (jchanam) [inline operator/redisfailover/service/heal.go]: You have a typo here. This should be "SetRoleLabel"

File: operator/redisfailover/service/constants.go [hunk 2]
```diff
@@ -40,4 +40,7 @@ const (
 	redisGroupName         = "mymaster"
 	appLabel               = "redis-failover"
 	hostnameTopologyKey    = "kubernetes.io/hostname"
+	redisRoleLabelName     = "RedisRole"
```
Comment (ese) [inline operator/redisfailover/service/constants.go]: Since `RedisRole` is quite redundant for the components of operator I would prefer it was `role`

**Comment by ese** [comment]
thanks @klnchu ! Only cosmetic stuff

**Comment by jchanam** [comment]
Sorry @klnchu,

I've checked the apimachinery api, and the call you're doing only [changes]() the labels on the object, locally.

```
func (meta *ObjectMeta) SetLabels(labels map[string]string)           { meta.Labels = labels }
```

[Here](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L158) you can see the function.

So, doing that won't change the Pod on K8S. Hence, this makes no sense to be done.

Thank you again!

**Comment by klnchu** [comment]
So sorry @jchanam,

I just used k8s, so I'm not familiar with it.

Have any other way to do that ?

File: operator/redisfailover/service/constants.go (hunk 2 above)
Comment (klnchu) [reply operator/redisfailover/service/constants.go]: It's OK.  

Thanks

**Comment by klnchu** [comment]
Before Time, I want to use label(master/slave) selector on service,  discovery the master/slave node.

then, I found "Sentinel  Service"  do this.

so, give it up.

Thanks.

**Comment by jchanam** [comment]
Yes, the correct way to connect to redis, is using Sentinel.

You connect to Sentinel and it'll give you which Redis to use. I recommend you using a proper library on your code, that allows using this method.

KR,

**Comment by so0k** [comment]
@klnchu - if you can not change client code, deploy client with sentinel-tunnel sidecar - https://github.com/RedisLabs/sentinel_tunnel

**Pull request closed without merging**